src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Operation/NewAzureVMCommand.cs

```diff
@@ -510,6 +521,29 @@ public void DefaultExecuteCmdlet()
             }
         }
 
+        /// <summary>
+        /// Heres whats happening here :
+        /// If "SystemAssignedIdentity" and "UserAssignedIdentity" are both present we set the type of identity to be SystemAssignedUsrAssigned and set the user 
+        /// defined identity in the VM identity object.
+        /// If only "SystemAssignedIdentity" is present, we just set the type of the Identity to "SystemAssigned" and no identity ids are set as its created by Azure
+        /// If only "UserAssignedIdentity" is present, we set the type of the Identity to be "UserAssigned" and set the Identity in the VM identity object.
+        /// If neither is present, we return a null.
+        /// </summary>
+        /// <returns>Returning the Identity generated form the cmdlet parameters "SystemAssignedIdentity" and "UserAssignedIdentity"</returns>
+        private VirtualMachineIdentity GetVMIdentityFromArgs()
+        {
+            var isUserAssignedEnabled = !string.IsNullOrWhiteSpace(UserAssignedIdentity);
+            return SystemAssignedIdentity.IsPresent || isUserAssignedEnabled
+                ? new VirtualMachineIdentity
+                {
+                    Type = !isUserAssignedEnabled ? CM.ResourceIdentityType.SystemAssigned
+                        : SystemAssignedIdentity.IsPresent ? CM.ResourceIdentityType.SystemAssignedUserAssigned
+                        : CM.ResourceIdentityType.UserAssigned,
```

Comment:
I had that initially, but this seems to be the preffered style here... @sergey-shandar comments?

Reply:
I do prefer `? :` because it's easy for me to read it as an expression, instead of procedure. For example, in procedural style (using if statement) it's really easy to miss one of the code-flow branch and forget to initialize the value. Another approach could be to use additional helper functions instead of nested `? :`.

Reply:
one ? : is easy to read, but nested ? : is not easy to read

Reply:
@hyonholee  I too agree with that, but the current case is not nested deep enough to warrant breaking up. I will restructure the code so that its laid out differently to make it more readable.